cpp/src/arrow/record_batch.cc

```diff
@@ -432,4 +433,43 @@ RecordBatchReader::~RecordBatchReader() {
   ARROW_WARN_NOT_OK(this->Close(), "Implicitly called RecordBatchReader::Close failed");
 }
 
+Result<std::shared_ptr<RecordBatch>> ConcatenateRecordBatches(
+    const RecordBatchVector& batches, MemoryPool* pool) {
+  int64_t length = 0;
+  size_t n = batches.size();
+  if (n == 0) {
+    return Status::Invalid("Must pass at least one recordbatch");
+  }
+  if (n == 1) {
```

Comment:
Instead, please make this deep copy the batch to conform with ConcatenateArrays https://github.com/apache/arrow/issues/37878

Reply:
Can it be changed to this method?
Within the col loop, the length of col=0 is assigned, and the lengths of the subsequent columns are verified against the first column.

```cpp
if (col == 0) {
   length = array->length();
} else if (length != array->length()) {
   return Status::Invalid(
            "column index ", i, " length is ", array->length(),
            ", does not match the length of previous columns: ", length);
}
```

Reply:
For a single batch, I removed this processing and let it follow the normal logic below